### PR TITLE
Generate a machine config name with the numerical prefix

### LINKED ISSUE #490: Prepend numeral index when generating performance profile MachineConfig
Currently, the operator generates a MachineConfig with the name of `performance-<pao-name>` and does not follows a convention of pretending a numerical value to the name so when the rendered version of the MCP is generated it is applied in whatever alphanumerical order it shows in the list of `oc get mc`. Now, when a cluster-admin wants to do further customizations, it cannot longer create MC with a prepend index like `99-my-custom-mc-fix` as it will not be applied correctly when the MCP is rendered.

My proposal is to at least pre-pend something like a 50 for having something like`50-performance-<pao-name>`

### PR DESCRIPTION
**Pull request by cynepco3hahue**
Currently, the operator generates a MachineConfig with the name of performance-<pao-name> and
does not follows a convention of pretending a numerical value to the name so when the rendered
version of the MCP is generated it is applied in whatever alphanumerical order it shows in the
list of oc get mc.
Now, when a cluster-admin wants to do further customizations, it cannot longer create MC with
a prepend index like 99-my-custom-mc-fix as it will not be applied correctly when the MCP is rendered.

Fixes #490

Signed-off-by: Artyom Lukianov <alukiano@redhat.com>

### CODE REVIEW

**Comment by MarSik** [comment]
/hold

I believe simply changing the generated name might break upgrade so before we merge this, please make sure you describe the flow and test it!

**Review by fromanirh — APPROVED** [review]
/lgtm

**Pull request merged**
